### PR TITLE
Cherry pick autocomplete fixes to tensorflow

### PR DESCRIPTION
**Pull request by marcrasi**
Cherry-picks some PRs necessary for https://bugs.swift.org/browse/SR-12076 into `tensorflow`.

This PR is only cherry-picks and llvm-project commit hash update, no additional changes.

Full explanation of the fix is in the PR description of the corresponding llvm-project PR: https://github.com/apple/llvm-project/pull/751.

### CODE REVIEW

File: utils/update_checkout/update-checkout-config.json [hunk 1]
```diff
@@ -308,7 +308,7 @@
        "tensorflow": {
             "aliases": ["tensorflow"],
             "repos": {
-                "llvm-project": "13c814f63ed68657fb480dfa428c94cbe82fc064",
+                "llvm-project": "8d389494ae739c7c14dfed9b554ebe0993014568",
```
Comment (ematejska) [inline utils/update_checkout/update-checkout-config.json]: Won't you need to update this with the following change https://github.com/apple/llvm-project/pull/751?

**Review by ematejska — APPROVED** [review]
Cherry picks look fine, just one question about the hash update.

File: utils/update_checkout/update-checkout-config.json (hunk 1 above)
Comment (marcrasi) [reply utils/update_checkout/update-checkout-config.json]: Just merged that, and updated this to the corresponding commit hash.

**Comment by marcrasi** [comment]
@swift-ci please test tensorflow

**Comment by marcrasi** [comment]
@swift-ci please clean test tensorflow

**Pull request merged**